ts/IdentificationModal.tsx

```diff
@@ -0,0 +1,191 @@
+import { Button, Content, Text, View } from "native-base";
+import * as React from "react";
+import { Modal, StatusBar } from "react-native";
+import { connect } from "react-redux";
+
+import Pinpad from "./components/Pinpad";
+import BaseScreenComponent from "./components/screens/BaseScreenComponent";
+import IconFont from "./components/ui/IconFont";
+import TextWithIcon from "./components/ui/TextWithIcon";
+import I18n from "./i18n";
+import {
+  identificationCancel,
+  identificationFailure,
+  identificationPinReset,
+  identificationSuccess
+} from "./store/actions/identification";
+import { ReduxProps } from "./store/actions/types";
+import { IdentificationState } from "./store/reducers/identification";
+import { GlobalState } from "./store/reducers/types";
+
+type MapStateToProps = {
+  identification: IdentificationState;
+};
+
+type Props = MapStateToProps & ReduxProps;
+
+type IdentificationByPinState = "unstarted" | "failure";
```

Comment:
I don't understand the difference between `IdentificationState` and `IdentificationByPinState`, could you add some comments?

Reply:
Yes going to add comments. For the moment we have only PIN identification but in future we'll have also other identification methods (like fingerprint). IdentificationState is to store the global user identification. IdentificationByPinState is a local state to check the result of the matching in Pinpad component.